src/components/application_manager/src/state_controller_impl.cc

```diff
@@ -511,12 +511,20 @@ bool StateControllerImpl::IsStateAvailable(ApplicationSharedPtr app,
     return IsStateAvailableForResumption(app, state);
   }
 
-  if (IsTempStateActive(HmiState::StateID::STATE_ID_AUDIO_SOURCE) ||
-      IsTempStateActive(HmiState::StateID::STATE_ID_EMBEDDED_NAVI)) {
-    if (HMILevel::HMI_FULL == state->hmi_level()) {
+  if (HMILevel::HMI_FULL == state->hmi_level()) {
+    if (IsTempStateActive(HmiState::StateID::STATE_ID_AUDIO_SOURCE) &&
+        app->is_media_application()) {
       LOG4CXX_DEBUG(logger_,
-                    "AUDIO_SOURCE or EMBEDDED_NAVI is active."
-                        << " Requested state is not available");
+                    "Media app is not allowed to activate due"
+                        << " to AUDIO_SOURCE event is active");
+      return false;
+    }
+
+    if (IsTempStateActive(HmiState::StateID::STATE_ID_EMBEDDED_NAVI) &&
+        app->IsVideoApplication()) {
```

Comment:
@jacobkeeler not sure about that. From one side you are right that projection application shows its own content which may be not related to navigation. From another side, when navigation is active there is no sense to show visual content of another application (actually I can't imagine the case when embedded navigation and projection app content can be displayed at the same time).
Do you have any ideas for that?

Reply:
@AKalinich-Luxoft I can't imagine that any app could be displayed at the same time as embedded navigation. Either an app can be activated during the EMBEDDED_NAVIGATION event or it can't, I don't see any reason that PROJECTION apps should be treated differently.

Looking at the documentation for the events, perhaps we should follow this chart: https://smartdevicelink.com/en/guides/hmi/basiccommunication/oneventchanged/#hmi-status-of-apps-when-audio_source-or-embedded_navi-event-is-activated

Meaning that only media (and perhaps communication) apps can be activated during EMBEDDED_NAVI and only navi apps can be activated during AUDIO_SOURCE. Notably both cases involve these apps activating to LIMITED, rather than FULL.

Reply:
Also, as a point of clarification, are the EMBEDDED_NAVIGATION and AUDIO_SOURCE events only active when these pieces of the HMI are in the foreground, or are they still active when you move to another screen? It seems to me if the user selects a media app while embedded audio is active, the app should take over audio, and navigation should be similar.